doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,299 @@
+specific ‘sequence’ parameter which incorporates the policies.
+Once the application has decided which policies to use, it can craft a FABRID HBH extension and include this as an option when sending
+the packet.
+
+DRKey
+^^^^^^
+
+FABRID uses DRKey for computing the Encrypted Policy Indices, the FABRID Hop Validation Fields and the Path Validator.
+The routers use the fast key derivation side, whereas the endhosts will use the slow side.
+
+Configuration
+--------------
+
+Control service
+^^^^^^^^^^^^^^^^^^
+
+To be able to use DRKey, one has to configure the control service setting "drkey.level1_db" and "drkey.secret_value_db".
+Additionally, since the border routers will fetch the secret value from the control service, the control service also has to
+add the internal IP address of all border routers of the local AS to the DRKey delegation list for FABRID.
+
+This could look like this::
+
+    [drkey.level1_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-level1.db"
+
+    [drkey.secret_value_db]
+    connection = "gen-cache/cs1-ff00_0_110-1.drkey-secret.db"
+
+    [drkey.delegation]
+    FABRID = [ "fd00:f00d:cafe::7f00:11", "fd00:f00d:cafe::7f00:12", "fd00:f00d:cafe::7f00:13"]
+
+The FABRID policies are configured in the control service. TODO(jelte): add more details
+
+Border router
+^^^^^^^^^^^^^^^
+
+For a router to query the DRKey secret value from the control service, once has to enable this.
+
+This could look like this::
+
+    [router]
+    use_drkey = true
+
+Considerations for future work
+--------------------------------
+
+SCMP response
+^^^^^^^^^^^^^^^
+
+With the current implementation, the sending endhost is not being informed when his packet gets dropped due to a FABRID error.
+In the future the border routers might send an SCMP response if they encounter an error when processing FABRID which might
+help the sending endhost in figuring out why his packet does not arrive at its destination.
+
+Rationale
+==========
+
+Path type vs HBH extension
+--------------------------------
+
+FABRID can be implemented either as a HBH extension or a path type.
+The reason why we decided against a path type is that FABRID as a HBH extension is incrementally deployable, whereas
+a new path type is not.
+
+Identifier option vs include everything in FABRID option
+------------------------------------------------------------
+
+We decided to move the packet ID and packet timestamp to another HBH option, the so called Identifier option,
+because this might also be useful for other HBH extensions and not just for FABRID (e.g., it would allow to port EPIC-HP from a path type to a HBH extension).
+Since FABRID still requires the packetID and packet timestamp, providing the Identifier option became mandatory for FABRID packets.
+
+Compatibility
+===============
+
+FABRID is a new extension which uses the SCION Hop-by-Hop extension which allows
```

Comment:
The "FABRID enabled" flag allows the sending endhost to choose for which ASes to enable path validation, but is also specifically required, for edge-cases where an AS just starts to roll out FABRID functionality to its BRs, but the sender does not yet know that this AS is FABRID aware. Because if a packet does not contain a HVF for this AS, the packet has to be dropped, unless the sender can explicitly state, that FABRID is disabled for this hop.

Reply:
Thanks, I added this part